questions/R_Demos/test_R_function/tests/test_03_correct_df.R

```diff
@@ -8,14 +8,15 @@ using(ttdo)
 
 file <- "/grade/serverFilesCourse/datasets/penguins.csv"
 
+## Get student's result
+student <- plr::source_and_eval_safe_with_hiding("/grade/student/student1.R",
+                                                 penguin_reader(file),
+                                                 "ag",
+                                                 "/grade/tests/solution.R")
+
 ## Get correct result
 source("/grade/tests/ans.R")
 reference <- penguin_read_reference(file)
 
-## Get student's result
-student <- plr::source_and_eval_safe("/grade/student/student1.R",
-                                     penguin_reader(file),
-                                     "ag")
-
```

Comment:
I vaguely recall a preference for 'correct result first' before trying student code which may (accidentally) have side effects.  Is there a deeper reason you re-ordered?

Reply:
I honestly forget the reason why, because I think the better defense is to not name student and solution objects with the same name. (Which I believe the object question does correctly.) But also, if you source the solution file first, the student code could have access to objects or functions from that, which could in some situations result in a "blank" student submission passing tests depending on how tests are written.

So I think the only "valid" reason for this change is that it now mirrors the order in the object question.

Reply:
Yup that argument is good too for the changed order -- let's keep this suggested change.